hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -371,6 +381,7 @@ def _networks(bits):
         # memoryview types, it is a subclass of Hashable and those types are not.
         # We therefore only generate the bytes type.
         typing.ByteString: st.binary(),
+        collections.abc.ByteString: st.binary(),
```

Comment:
What about `AnyStr`? Will it be handled via `TypeVar`?

Reply:
`AnyStr is TypeVar[str, bytes]` :grin:

Reply:
Let's add a test case for it, maybe? Currently it is not tested https://github.com/HypothesisWorks/hypothesis/search?q=AnyStr

Reply:
We do test TypeVar though, including via `typing.Pattern` (which is implicitly `Pattern[AnyStr]`).  Since these `from_type()` tests are relatively slow I'd rather not add more than we really need to catch regressions.